src/main/java/org/broadinstitute/hdf5/HDF5File.java

```diff
@@ -477,21 +510,21 @@ private void checkIsOpen() {
     /**
      * Opens a HDF5 file given its access {@link OpenMode mode}.
      */
-    private static int open(final File file, final OpenMode mode) {
+    private static int open(final Path file, final OpenMode mode) {
         final int fileId;
         try {
             if (mode == OpenMode.CREATE) {
-                file.delete();
-                file.createNewFile();
+                Files.deleteIfExists(file);
+                Files.createFile(file);
             }
-            fileId = H5.H5Fopen(file.getAbsolutePath(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
+            fileId = H5.H5Fopen(file.toAbsolutePath().toString(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
```

Comment:
I would like to have support to write HD5 files too, but it is true that it could be very messy if it is on close. I will think about it and come back to the PR if I could find a solution...if not, I will add the cached-reading for having at least some support...